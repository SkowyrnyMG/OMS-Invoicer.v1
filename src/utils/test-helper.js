import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
