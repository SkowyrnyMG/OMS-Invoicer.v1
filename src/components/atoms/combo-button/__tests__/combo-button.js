import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import ComboButton from '../combo-button';

describe('ComboButton', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<ComboButton />);
  });
});
