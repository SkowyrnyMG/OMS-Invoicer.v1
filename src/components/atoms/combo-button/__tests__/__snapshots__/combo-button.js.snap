// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboButton should display in the document with correct styles 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  border: 2px solid #446DF6;
  border-radius: 0.5rem;
  background: none;
  cursor: pointer;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c0:not(:first-child) {
  margin-left: 2rem;
}

.c0:hover {
  background: #446DF6;
  color: #FFFFFF;
}

<button
  class="c0"
/>
`;
